### ISSUE TITLE
false error on `.either`

### ISSUE THREAD

**Issue by strokyl** [zio/zio-intellij#490]
I got the following code:
```
ZIO.serviceWithZIO[IndexedTopicService](_.list(aCluster.clusterSlug, apiCaller)).either
```
highlighted in red with the following error:
```
Replace with .map(Right(_)) 
Effect cannot fail; operation .either is impossible
```

To signature of the list method of IndexedTopicService:
```
  def list(
    clusterName: values.ClusterSlug,
    caller: core.domain.auth.ApiCaller
  ): IO[ConsoleError, List[IndexedTopic]]
```
Removing the ZIO plugin, remove the error.

ZIO plugin version: 2024.3.35.0
Intelij version: 2024.4.3
Scala: 3.5.2

The project is imported using BSP.

**Comment by hmemcpy** [comment]
Ugh, sorry about that. I'll take a look.

**Comment by hmemcpy** [comment]
Yeah, just as I expected. This is, unfortunately, a [bug in the Scala plugin](https://youtrack.jetbrains.com/issue/SCL-23214/Partially-applied-overload-is-resolved-incorrectly) that made it into the release version. It breaks type signatures for partially-applied types. See this little squiggly?

<img width="353" alt="image" src="https://github.com/user-attachments/assets/fa5bfe33-30f6-4343-9f10-fa061307ac93">

This is IntelliJ resolving the signature of `serviceWithZIO` to the signature of its `apply` method instead, thinking it's missing some type arguments. This breaks the ZIO plugin because it now fails to correctly infer the environment.

Anyway, the team said they're working on a fix and it should be available soon, and hopefully this just "fixes" the ZIO plugin issue as well.

Keeping this issue open to verify when the fix becomes available.

**Comment by strokyl** [comment]
Thank you for the information

**Comment by myazinn** [comment]
Hi!
The fix is available and seems to work fine now (for quite a while actually, sorry for the delayed response)
